### PR TITLE
Refactor threading into ParallelValidator

### PR DESCRIPTION
**Pull request by briantkennedy**
This change is intended to simplify Validator to enable emitting results
in a format other than Violation.

### CODE REVIEW

File: pkg/gcv/parallelvalidator.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+type assetResult struct {
+	violations []*validator.Violation
+	err        error
+}
+
+// NewParallelValidator creates a new instance with the given stop channel and validator
+func NewParallelValidator(stopChannel <-chan struct{}, cv ConfigValidator) *ParallelValidator {
+	pv := &ParallelValidator{
+		work: make(chan func(), flags.workerCount*2),
```
Comment (t12g) [inline pkg/gcv/parallelvalidator.go]: can we add a comment to explain why the channel buffer size is 2x worker count?

File: pkg/gcv/parallelvalidator.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+			return &assetResult{violations: violations}
+		}()
+	}
+}
+
+// Review evaluates each asset in the review request in parallel and returns any
+// violations found.
+func (v *ParallelValidator) Review(ctx context.Context, request *validator.ReviewRequest) (*validator.ReviewResponse, error) {
+	assetCount := len(request.Assets)
+	resultChan := make(chan *assetResult, flags.workerCount*2)
```
Comment (t12g) [inline pkg/gcv/parallelvalidator.go]: should the size of resultChan match that of ParallelValidator.work? if so, can we add a comment?

File: pkg/gcv/parallelvalidator.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
+package gcv
+
+import (
+	"context"
+	"flag"
+	"runtime"
+
+	"github.com/pkg/errors"
+
```
[the remark this answers is not in the document]
Comment (briantkennedy) [reply pkg/gcv/parallelvalidator.go]: weird, unsure how those got in there, might be some intellij thing.

File: pkg/gcv/parallelvalidator.go (hunk 1 above)
Comment (briantkennedy) [reply pkg/gcv/parallelvalidator.go]: added

File: pkg/gcv/parallelvalidator.go (hunk 2 above)
Comment (briantkennedy) [reply pkg/gcv/parallelvalidator.go]: done

**Pull request merged**